src/Templates/MethodTemplate.cshtml

```diff
@@ -177,44 +209,71 @@ func (client @(Model.Owner)) @(Model.ResponderMethodName)(resp *http.Response) (
         @(Model.RespondDecorators.EmitAsArguments()))
 
     @(Model.Response(true))
+</text>
+}
     return
 }
 
+@if (Model.IsLongRunningOperation() && Model.IsPageable)
+{
+    var pageType = Model.ReturnType.Body.Cast<CompositeTypeGo>().UnwrapPageType();
+<text>
+@EmptyLine
+func (client @(Model.Owner)) @(Model.ResponderMethodName.ToCamelCase())(resp *http.Response) (result @pageType.ContentType.Name, err error) {
+    @if (Model.ReturnValue().Body.IsStreamType())
+    {
+    @:result.Value = &resp.Body
+    }
+    err = autorest.Respond(
+        @(Model.RespondDecorators.EmitAsArguments()))
+
+    @(Model.Response(true))
+    return
+}
+</text>
+}
+
 @if (Model.IsPageable && !Model.NextAlreadyDefined)
 {
-    var lastResTypyeName = Model.ReturnValue().Body.Name;
-    var preparerName = Model.ReturnValue().Body.Cast<CompositeTypeGo>().PreparerMethodName;
+    var preparerName = Model.ReturnValue().Body.Cast<CompositeTypeGo>().UnwrapPageType().PreparerMethodName;
     if (Model.IsLongRunningOperation())
     {
-        lastResTypyeName = Model.ReturnValue().Body.Cast<FutureTypeGo>().ResultTypeName;
-        preparerName = $"{lastResTypyeName}Preparer";
+        preparerName = $"{Model.LastResultsTypeName().ToCamelCase()}Preparer";
     }
 <text>
 @EmptyLine
 // @(Model.NextMethodName) retrieves the next set of results, if any.
-func (client @(Model.Owner)) @(Model.NextMethodName)(lastResults @lastResTypyeName) (@Model.MethodReturnSignature()) {
+func (client @(Model.Owner)) @(Model.NextMethodName)(lastResults @Model.LastResultsTypeName()) (@Model.NextMethodReturnSignature()) {
```

Comment:
>)(lastResults @Model.LastResultsTypeName() [](start = 51, length = 42)

Not 100% sure why we need to pass the `lastResults` as a parameter here.

Reply:
is it just to get the preparer ? Im trying to think if there is a better way...

---
In reply to: [154745910](https://github.com/Azure/autorest.go/pull/59#discussion_r154745910) [](ancestors = 154745910)

Reply:
although the `listNextResults` is a global command so I guess it needs to know the next results of whom to get, so seems to be fine ( also is not exported )

---
In reply to: [154746103](https://github.com/Azure/autorest.go/pull/59#discussion_r154746103) [](ancestors = 154746103,154745910)

Reply:
We call the generated preparer on it, see ModelTemplate.cshtml.